presto-main/src/main/java/com/facebook/presto/operator/aggregation/noisyaggregation/NoisyCountGaussianColumnRandomSeedAggregation.java

```diff
@@ -0,0 +1,161 @@
+import static com.facebook.presto.common.type.DoubleType.DOUBLE;
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.operator.aggregation.AggregationUtils.generateAggregationName;
+import static com.facebook.presto.operator.aggregation.noisyaggregation.NoisyCountGaussianColumnAggregationUtils.computeNoisyCount;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.spi.function.Signature.typeVariable;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.BLOCK_INDEX;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.BLOCK_INPUT_CHANNEL;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.NULLABLE_BLOCK_INPUT_CHANNEL;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.STATE;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public class NoisyCountGaussianColumnRandomSeedAggregation
+        extends SqlAggregationFunction
+{
+    public static final NoisyCountGaussianColumnRandomSeedAggregation NOISY_COUNT_GAUSSIAN_RANDOM_SEED_AGGREGATION = new NoisyCountGaussianColumnRandomSeedAggregation();
+    private static final String NAME = "noisy_count_gaussian";
+    private static final MethodHandle INPUT_FUNCTION = methodHandle(NoisyCountGaussianColumnRandomSeedAggregation.class, "input", CountScaleRandomSeedState.class, Block.class, Block.class, Block.class, int.class);
+    private static final MethodHandle COMBINE_FUNCTION = methodHandle(NoisyCountGaussianColumnRandomSeedAggregation.class, "combine", CountScaleRandomSeedState.class, CountScaleRandomSeedState.class);
+    private static final MethodHandle OUTPUT_FUNCTION = methodHandle(NoisyCountGaussianColumnRandomSeedAggregation.class, "output", CountScaleRandomSeedState.class, BlockBuilder.class);
+
+    public NoisyCountGaussianColumnRandomSeedAggregation()
+    {
+        super(NAME,
+                ImmutableList.of(typeVariable("T")),
+                ImmutableList.of(),
+                parseTypeSignature(StandardTypes.BIGINT),
+                ImmutableList.of(parseTypeSignature("T"), DOUBLE.getTypeSignature(), BIGINT.getTypeSignature()));
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Counts the non-null values and then add Gaussian noise to the true count. The noisy count is post-processed to be non-negative and rounded to bigint. Random seed is used to seed random generator when it is positive. If random seed is <= 0, no seed is provided to the random generator. Not using a secure random.";
```

Comment:
> If random seed is <= 0, no seed is provided to the random generator. Not using a secure random.

This seems unclear/contradictory the actual implementation in `output()`, which does use `SecureRandom` when `seed <= 0`, though see comment below.

Reply:
Yeah, the text is a bit hard to read, but the idea is the same: if seed > 0 is provided, not using secure random.